### PR TITLE
Xonsh Activation fixes

### LINKED ISSUE #8274: conda init does nothing for xonsh
## Current Behavior

I'm using xonsh on ubuntu. I first tried to `conda activate base` (or some other environment) in but then I get an error and suggestion to run `conda init`. Next I run `conda init` without error, but no change is reported and I cannot get `conda activate base` to work.

### Steps to Reproduce

```xonsh
λ which conda
/home/pletnes/miniconda3/bin/conda
λ conda activate base
CommandNotFoundError: Your shell has not been properly configured to use 'conda activate'. 
To initialize your shell, run
$ conda init
Currently supported shells are:
  - bash
  - fish
  - tcsh
  - xonsh
  - zsh
  - powershell
See 'conda init --help' for more information and options.
IMPORTANT: You may need to close and restart your shell after running 'conda init'.

λ conda init xonsh
no change     /home/pletnes/miniconda3/condabin/conda
no change     /home/pletnes/miniconda3/bin/conda
no change     /home/pletnes/miniconda3/bin/conda-env
no change     /home/pletnes/miniconda3/bin/activate
no change     /home/pletnes/miniconda3/bin/deactivate
no change     /home/pletnes/miniconda3/etc/profile.d/conda.sh
no change     /home/pletnes/miniconda3/etc/fish/conf.d/conda.fish
no change     /home/pletnes/miniconda3/shell/condabin/Conda.psm1
no change     /home/pletnes/miniconda3/shell/condabin/conda-hook.ps1
no change     /home/pletnes/miniconda3/lib/python3.7/site-packages/xonsh/conda.xsh
no change     /home/pletnes/miniconda3/etc/profile.d/conda.csh

λ conda activate base
xonsh: For full traceback set: $XONSH_SHOW_TRACEBACK = True
ModuleNotFoundError: No module named 'conda'
```

## Expected Behavior

I hoped `conda init xonsh` would create the necessary PATH (or similar) settings in `~/.xonshrc` so that `conda activate <some-env>` works flawlessly.

## Environment Information
<details open><summary><code>`conda info`</code></summary><p>

```
     active environment : base
    active env location : /home/pletnes/miniconda3
            shell level : 1
       user config file : /home/pletnes/.condarc
 populated config files :
          conda version : 4.6.3
    conda-build version : not installed
         python version : 3.7.1.final.0
       base environment : /home/pletnes/miniconda3  (writable)
           channel URLs : https://repo.anaconda.com/pkgs/main/linux-64
                          https://repo.anaconda.com/pkgs/main/noarch
                          https://repo.anaconda.com/pkgs/free/linux-64
                          https://repo.anaconda.com/pkgs/free/noarch
                          https://repo.anaconda.com/pkgs/r/linux-64
                          https://repo.anaconda.com/pkgs/r/noarch
          package cache : /home/pletnes/miniconda3/pkgs
                          /home/pletnes/.conda/pkgs
       envs directories : /home/pletnes/miniconda3/envs
                          /home/pletnes/.conda/envs
               platform : linux-64
             user-agent : conda/4.6.3 requests/2.21.0 CPython/3.7.1 Linux/4.4.0-142-generic ubuntu/16.04.5 glibc/2.23
                UID:GID : 3506278:17
             netrc file : None
           offline mode : False
```

And xonsh version:
```
$ xonsh --version    
xonsh/0.8.10
```
</p></details>

<details open><summary><code>`conda config --show-sources`</code></summary><p>

```
$ conda config --show-sources
# this returned blank?
```
</p></details>

<details><summary><code>`conda list --show-channel-urls`</code></summary><p>

```
$ conda list --show-channel-urls
# packages in environment at /home/paul/miniconda3:
#
# Name                    Version                   Build  Channel
asn1crypto                0.24.0                   py37_0    defaults
ca-certificates           2019.1.23                     0    defaults
certifi                   2018.11.29               py37_0    defaults
cffi                      1.11.5           py37he75722e_1    defaults
chardet                   3.0.4                    py37_1    defaults
conda                     4.6.3                    py37_0    defaults
conda-env                 2.6.0                         1    defaults
cryptography              2.4.2            py37h1ba5d50_0    defaults
idna                      2.8                      py37_0    defaults
libedit                   3.1.20170329         h6b74fdf_2    defaults
libffi                    3.2.1                hd88cf55_4    defaults
libgcc-ng                 8.2.0                hdf63c60_1    defaults
libstdcxx-ng              8.2.0                hdf63c60_1    defaults
ncurses                   6.1                  he6710b0_1    defaults
openssl                   1.1.1a               h7b6447c_0    defaults
pip                       18.1                     py37_0    defaults
pycosat                   0.6.3            py37h14c3975_0    defaults
pycparser                 2.19                     py37_0    defaults
pyopenssl                 18.0.0                   py37_0    defaults
pysocks                   1.6.8                    py37_0    defaults
python                    3.7.1                h0371630_7    defaults
readline                  7.0                  h7b6447c_5    defaults
requests                  2.21.0                   py37_0    defaults
ruamel_yaml               0.15.46          py37h14c3975_0    defaults
setuptools                40.6.3                   py37_0    defaults
six                       1.12.0                   py37_0    defaults
sqlite                    3.26.0               h7b6447c_0    defaults
tk                        8.6.8                hbc83047_0    defaults
urllib3                   1.24.1                   py37_0    defaults
wheel                     0.32.3                   py37_0    defaults
xz                        5.2.4                h14c3975_4    defaults
yaml                      0.1.7                had09818_2    defaults
zlib                      1.2.11               h7b6447c_3    defaults
```
</p></details>

### PR DESCRIPTION
**Pull request by scopatz**
This fixes the xonsh activation tools so that they should actually work. I have tested it out in the dev environment.

CC @gforsyth @teoliphant 

fixes #8274

### CODE REVIEW

**Comment by soapy1** [comment]
fixes https://github.com/conda/conda/issues/8274

**Comment by scopatz** [comment]
Wow, it actually passed CI too!

File: tests/core/test_initialize.py [hunk 1]
```diff
@@ -295,7 +295,7 @@ def test_make_install_plan(self):
                         "function": "install_conda_xsh",
                         "kwargs": {
                             "conda_prefix": "/darwin",
-                            "target_path": "/darwin/lib/python2.6/site-packages/xonsh/conda.xsh"
+                            "target_path": "/darwin/lib/python2.6/site-packages/xontrib/conda.xsh"
```
Comment (gforsyth) [inline tests/core/test_initialize.py]: wait, what?

Comment (scopatz) [reply tests/core/test_initialize.py]: Yeah, conda will let you stick these files anywhere, so I figured `xontrib/` was the right place for it.

Comment (gforsyth) [reply tests/core/test_initialize.py]: I was more curious about the `lib/python2.6` part of it

**Comment by gforsyth** [comment]
Not sure what's going on here because you didn't touch this file, but trying to set up the dev environment throws this error on my machine:

```
Traceback (most recent call last):
  File "/home/gil/github.com/scopatz/conda/devenv/lib/python3.7/runpy.py", line 193, in _run_module_as_main
    "__main__", mod_spec)
  File "/home/gil/github.com/scopatz/conda/devenv/lib/python3.7/runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "/home/gil/github.com/scopatz/conda/conda/__main__.py", line 8, in <module>
    sys.exit(main())
  File "/home/gil/github.com/scopatz/conda/conda/cli/main.py", line 150, in main
    return conda_exception_handler(_main, *args, **kwargs)
  File "/home/gil/github.com/scopatz/conda/conda/exceptions.py", line 1305, in conda_exception_handler
    return_value = exception_handler(func, *args, **kwargs)
  File "/home/gil/github.com/scopatz/conda/conda/exceptions.py", line 1020, in __call__
    return self.handle_exception(exc_val, exc_tb)
  File "/home/gil/github.com/scopatz/conda/conda/exceptions.py", line 1064, in handle_exception
    return self.handle_unexpected_exception(exc_val, exc_tb)
  File "/home/gil/github.com/scopatz/conda/conda/exceptions.py", line 1075, in handle_unexpected_exception
    self.print_unexpected_error_report(error_report)
  File "/home/gil/github.com/scopatz/conda/conda/exceptions.py", line 1145, in print_unexpected_error_report
    from .cli.main_info import get_env_vars_str, get_main_info_str
  File "/home/gil/github.com/scopatz/conda/conda/cli/main_info.py", line 19, in <module>
    from ..core.index import _supplement_index_with_system
  File "/home/gil/github.com/scopatz/conda/conda/core/index.py", line 10, in <module>
    from .package_cache_data import PackageCacheData
  File "/home/gil/github.com/scopatz/conda/conda/core/package_cache_data.py", line 39, in <module>
    from ..utils import human_bytes
  File "/home/gil/github.com/scopatz/conda/conda/utils.py", line 133, in <module>
    path_prepend=dedent("""\
NameError: name 'dedent' is not defined
```

File: tests/core/test_initialize.py (hunk 1 above)
Comment (scopatz) [reply tests/core/test_initialize.py]: Oh yeah, I think that is a meaningless PREFIX sting.  Conda itself doesn't support 2.6 either

**Comment by scopatz** [comment]
I saw that at some point to.  I had to run `dev/start` from Bash.

**Comment by gforsyth** [comment]
> I saw that at some point to. I had to run `dev/start` from Bash.

hrm, that was in bash.  I'll try again.

**Comment by msarahan** [comment]
It seems that the circle CI tests didn't run.  I'm going to close/reopen this and hope that it comes back.

**Comment by scopatz** [comment]
Weird, circle didn't run this time either...

**Comment by msarahan** [comment]
*shrug* - happy to merge and sort it out later.

**Comment by scopatz** [comment]
Yay! Thanks!

**Pull request merged**